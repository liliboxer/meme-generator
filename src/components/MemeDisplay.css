.MemeDisplay {
position: relative;
text-align: center;
width: 50vh;

img {
  width: 100%;
}

   .centeredTop {
    position: absolute;
    text-align: center;
    width: 100%;
    top: 10px;
    left: 10px;
    text-shadow:
        -1px -1px 0 #000,
        1px -1px 0 #000,
        -1px 1px 0 #000,
        1px 1px 0 #000;
   }

   .centeredBottom {
    position: absolute;
    text-align: center;
    width: 100%;
    bottom: 10px;
    left: 10px;
    text-shadow:
        -1px -1px 0 #000,
        1px -1px 0 #000,
        -1px 1px 0 #000,
        1px 1px 0 #000;
   }
}