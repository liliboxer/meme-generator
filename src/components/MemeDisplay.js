import React from 'react';
import PropTypes from 'prop-types';
import styles from './MemeDisplay.css';

function MemeDisplay({ imageURL, file, color, formattedText, font }) {
  return (
    <section className={styles.MemeDisplay} id="meme">
      <pre
        style={{ color: color } }
        className={styles.centeredTop}>
        {formattedText}
      </pre>
      <img src={imageURL || file } ></img>
      <pre
        style={{ 
          color: color,
          fontFamily: font
        }}
        className={styles.centeredBottom}>
        {formattedText}
      </pre>
    </section>
  );
}

MemeDisplay.propTypes = {
  imageURL: PropTypes.string.isRequired,
  file: PropTypes.oneOfType([null, PropTypes.object]),
  color: PropTypes.string.isRequired,
  formattedText: PropTypes.string.isRequired,
  font: PropTypes.string.isRequired
};

export default MemeDisplay;
