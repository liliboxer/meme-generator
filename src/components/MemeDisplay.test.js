import React from 'react';
import { shallow } from 'enzyme';
import MemeDisplay from './MemeDisplay';

describe('MemeDisplay component', () => {
  it('renders MemeDisplay', () => {
    const wrapper = shallow(<MemeDisplay/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
