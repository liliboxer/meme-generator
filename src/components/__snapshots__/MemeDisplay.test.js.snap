// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemeDisplay component renders MemeDisplay 1`] = `
<section
  className="MemeDisplay"
  id="meme"
>
  <pre
    className="centeredTop"
    style={
      Object {
        "color": undefined,
        "fontFamily": undefined,
      }
    }
  />
  <img />
  <pre
    className="centeredBottom"
    style={
      Object {
        "color": undefined,
        "fontFamily": undefined,
      }
    }
  />
</section>
`;
